### ISSUE TITLE
Sbt console truncates long strings

### ISSUE THREAD

**Issue by AndreasKostler** [ensime/emacs-sbt-mode#21]
Sbt console truncates:
- sbt-send-region of this snippet will cause the sbt-console to truncate the output

``` scala
class ThisMightBreak {
  def aVeryLongLongLongLongLongLongLongLongLongLongLongLongMethodName () {}
}
```

results in (sbt console):

```
scala> class ThisMightBreak {
{}
     | }
defined class ThisMightBreak

scala> 
```

**Comment by fommil** [comment]
@AndreasKostler would you like to look into this further and send a PR?

I'm tempted to suggest deprecating the `console` support in favour of the ensime REPL (which is build tool agnostic) and doesn't require the user to start up an ensime server, so extremely lightweight.

**Comment by hvesalai** [comment]
Hmm... deprecating `console` support doesn't sound very nice.

**Comment by fommil** [comment]
we should perhaps move the `console` support into `scala-mode` then and share it between here and ensime? Seems silly to have two implementations of the same thing.

**Comment by hvesalai** [comment]
That also doesn't sound very good since at the moment scala-mode has no process interaction and adding it there would need all the boiler plate to be copied there. I quite don't understand the problem we are trying to fix by moving it anywhere.

This is sbt-mode so it should contain all things `sbt`. If somebody doesn't want to use the console-feature, then they just dont't.

**Comment by fommil** [comment]
It is the code duplication. it'd be good if Scala repl support lived in one place. A lot of the bug reports I'm seeing here are relevant for the ensime repl too.

**Comment by hvesalai** [comment]
Ah, so we have to repl (console) modes? Then why not merge them to gether and make it configurable which backend they use (new sbt session, existing sbt session, new ensime repl session).

**Comment by fommil** [comment]
Yes, exactly :-)

**Comment by ashawley** [comment]
Fixed already in #93?

**Comment by hvesalai** [comment]
Definitely fixed by turning echo off #134, see also #136
